File: src/main/resources/static/js/filter.js
'use strict';


const filter = (() => {
    let PLZ = "";
    let priceMin = "";
    let priceMax = "";
    let roomsMin = "";
    let roomsMax = "";

    return {
        getPLZ:      () => PLZ,
        getpriceMin: () => priceMin,
        getPriceMax: () => priceMax,
        getRoomsMin: () => roomsMin,
        getRoomsMax: () => roomsMax,
        setPlz:      newPlz      => PLZ      = newPlz,
        setPriceMin: newPriceMin => priceMin = newPriceMin,
        setPriceMax: newPriceMax => priceMax = newPriceMax,
        setRoomsMin:  newRoomsMin => roomsMin = newRoomsMin,
        setRoomsMax: newRoomsMax => roomsMax = newRoomsMax,
        getAllValues: () => [PLZ, priceMin, priceMax, roomsMin, roomsMax],
    }
})();

export {filter};

File: src/main/java/ch/propbuddy/propbuddy/domain/Filter.java
package ch.propbuddy.propbuddy.domain;

public class Filter {
    private final String location;
    private final String priceMin;
    private final String priceMax;
    private final String roomsMin;
    private final String roomsMax;

    public Filter(String location, String priceMin, String priceMax, String roomsMin, String roomsMax) {
        this.location = location;
        this.priceMin = priceMin;
        this.priceMax = priceMax;
        this.roomsMin = roomsMin;
        this.roomsMax = roomsMax;
    }

    public String getLocation() {
        return location;
    }

    public String getPriceMin() {
        return priceMin;
    }

    public String getPriceMax() {
        return priceMax;
    }

    public String getRoomsMin() {
        return roomsMin;
    }

    public String getRoomsMax() {
        return roomsMax;
    }

    @Override
    public String toString() {
        return location + "," + priceMin + "," + priceMax + "," + roomsMin + "," + roomsMax;
    }
}
